persist/sqlite/scan_test.go

```diff
@@ -121,52 +232,108 @@ func TestScan(t *testing.T) {
+		WriteBaseCost:                randSC(),
+		WriteLengthCost:              randSC(),
+		WriteStoreCost:               randSC(),
+		TxnFeeMinRecommended:         randSC(),
+		TxnFeeMaxRecommended:         randSC(),
+		ContractPrice:                randSC(),
+		CollateralCost:               randSC(),
+		MaxCollateral:                randSC(),
+
+		MaxDuration:          20,
+		WindowSize:           30,
+		RegistryEntriesLeft:  40,
+		RegistryEntriesTotal: 50,
+	}
+
+	v2Settings := proto4.HostSettings{
+		ProtocolVersion:     [3]uint8{4, 0, 0},
 		Release:             "test",
 		AcceptingContracts:  true,
 		WalletAddress:       w.Address(),
-		MaxCollateral:       types.Siacoins(10000),
+		MaxCollateral:       randSC(),
 		MaxContractDuration: 1000,
-		RemainingStorage:    100 * proto4.SectorSize,
-		TotalStorage:        100 * proto4.SectorSize,
+		RemainingStorage:    100,
+		TotalStorage:        100,
 		Prices: proto4.HostPrices{
-			ContractPrice: types.Siacoins(uint32(frand.Uint64n(10000))),
-			StoragePrice:  types.Siacoins(uint32(frand.Uint64n(10000))),
-			IngressPrice:  types.Siacoins(uint32(frand.Uint64n(10000))),
-			EgressPrice:   types.Siacoins(uint32(frand.Uint64n(10000))),
-			Collateral:    types.Siacoins(uint32(frand.Uint64n(10000))),
+			ContractPrice: randSC(),
+			StoragePrice:  randSC(),
+			IngressPrice:  randSC(),
+			EgressPrice:   randSC(),
+			Collateral:    randSC(),
 		},
-	})
+	}
+
+	sr := ctestutil.NewEphemeralSettingsReporter()
+	sr.Update(v2Settings)
 	ss := ctestutil.NewEphemeralSectorStore()
 	c := ctestutil.NewEphemeralContractor(cm)
 
-	// We do not have the private key to this host, so we copy their
-	// announcement data because we cannot sign an announcement as
-	// them.
-	var pubkey1 types.PublicKey
-	// pubkey of sia1.euregiohosting.nl:9982
-	if err := pubkey1.UnmarshalText([]byte(`ed25519:e89e13affe9d2ab4dc6f1e157376c60cdcadddf061ea78a52db68b63e6070ee4`)); err != nil {
-		t.Fatal(err)
-	}
-	// announcement copied from block 426479
-	announcement, err := base64.StdEncoding.DecodeString("SG9zdEFubm91bmNlbWVudBsAAAAAAAAAc2lhMS5ldXJlZ2lvaG9zdGluZy5ubDo5OTgyZWQyNTUxOQAAAAAAAAAAACAAAAAAAAAA6J4Tr/6dKrTcbx4Vc3bGDNyt3fBh6nilLbaLY+YHDuRQ49Qr0uy4nZJq3XVW/8kCsqnJFyQT4Zj/CcvC5fG5orVLwnVp2xYdn0nNvBdPNV9LzHborsYz4pep1ywwn40C")
-	if err != nil {
-		t.Fatal(err)
-	}
+	pk1 := types.GeneratePrivateKey()
+	pubkey1 := pk1.PublicKey()
 
 	pk2 := types.GeneratePrivateKey()
 	pubkey2 := pk2.PublicKey()
 
 	pk3 := types.GeneratePrivateKey()
 	pubkey3 := pk3.PublicKey()
 
-	v4Addr, _ := testRenterHostPair(t, pk3, cm, s, w, c, sr, ss, zap.NewNop())
+	pk4 := types.GeneratePrivateKey()
+	pubkey4 := pk4.PublicKey()
+
+	rhp2Addr, _ := testV1Host(t, pk1, &settings, &table)
+	v4Addr, _ := testV2Host(t, pk3, cm, s, w, c, sr, ss, zap.NewNop())
 
 	cfg := config.Scanner{
-		Threads:             10,
-		Timeout:             30 * time.Second,
+		BatchSize:           100,
+		Timeout:             10 * time.Second,
```

Comment:
This is still a very high number. The first thing `TestScan` does is wait 10s to sync up. Likely because of

```
func (e *Explorer) waitForSync() error {
	ticker := time.NewTicker(10 * time.Second)
	defer ticker.Stop()
```

`nit` in the logging I noticed `logger.go:146: 2025-03-19T09:14:05.779+0100 INFO    shutdown:       {"error": "context canceled"}`, which is not really a useful log line. If anything it could/should be at the `debug` level. In `renterd` we weren't great about this either and we get frequent complaints about the size of our `renterd.log`. In approach in indexd is to info log only when the error is actionable.

Reply:
Resolved in followup